virttest/utils_test/libvirt.py

```diff
@@ -1708,8 +1708,10 @@ def create_hostdev_xml(pci_id, boot_order=0):
         hostdev_xml.boot_order = boot_order
     attrs = {'domain': device_domain, 'slot': device_slot,
              'bus': device_bus, 'function': device_function}
-    hostdev_xml.source_address = hostdev_xml.new_source_address(**attrs)
+    hostdev_xml.source = hostdev_xml.new_source(**attrs)
     logging.debug("Hostdev XML:\n%s", str(hostdev_xml))
+    if not xmlfile:
+        return hostdev_xml
```

Comment:
This API is meant to return the Hostdev xml, why do we return the object ?

Reply:
For our test will need a hostdev object, and the pci_id parsing in this function is convinient.

Reply:
@chloerh sure, request you to add doc string that would explain the new param, as it helps.

Reply:
@chloerh even the name of method says only about xml, it is good to have a another method to have hostdev object,
if not otherwise a doc string helps...